tests/integration/java_export/BUILD

```diff
@@ -0,0 +1,60 @@
+load("//:defs.bzl", "artifact")
+load("//private/rules:maven_project_jar.bzl", "maven_project_jar")
+
+java_test(
+    name = "PublishShapeTest",
+    srcs = ["PublishShapeTest.java"],
+    test_class = "com.jvm.external.jvm_export.PublishShapeTest",
+    jvm_flags = [
+        "-Ddeploy.jar=$(location //private/tools/java/rules/jvm/external/maven:MavenPublisher_deploy.jar)",
+    ],
+    deps = [
+        artifact("com.google.guava:guava"),
+        artifact("org.hamcrest:hamcrest"),
+    ],
+    data = [
+        "//private/tools/java/rules/jvm/external/maven:MavenPublisher_deploy.jar",
+    ],
+)
+
+java_test(
+    name = "ProjectJarTest",
+    srcs = ["ProjectJarTest.java"],
+    test_class = "com.jvm.external.jvm_export.ProjectJarTest",
+    jvm_flags = [
+        "-Dlocations=\"$(locations :project)\"",
+    ],
+    deps = [
+        ":project",
+        artifact("com.google.guava:guava"),
+        artifact("org.hamcrest:hamcrest"),
+    ],
+    data = [
+        # It's not apparent why it's not sufficient to simply add this:
+        # one must also include this in the `deps` too, else the test
+        # can't find the file.
```

Comment:
`deps` adds the jar to the test's compile/runtime classpath, whereas `data` adds to the java-agnostic runfiles. `data` basically lets you call `Paths.get` in the test.